src/main/kotlin/app/revanced/patches/tumblr/timelinefilter/TimelineFilterPatch.kt

```diff
@@ -0,0 +1,68 @@
+package app.revanced.patches.tumblr.timelinefilter
+
+import app.revanced.extensions.exception
+import app.revanced.patcher.data.BytecodeContext
+import app.revanced.patcher.extensions.InstructionExtensions.addInstructions
+import app.revanced.patcher.extensions.InstructionExtensions.addInstructionsWithLabels
+import app.revanced.patcher.extensions.InstructionExtensions.removeInstructions
+import app.revanced.patcher.patch.BytecodePatch
+import app.revanced.patcher.patch.annotation.Patch
+import app.revanced.patches.tumblr.timelinefilter.fingerprints.PostsResponseConstructorFingerprint
+import app.revanced.patches.tumblr.timelinefilter.fingerprints.TimelineConstructorFingerprint
+import app.revanced.patches.tumblr.timelinefilter.fingerprints.TimelineFilterIntegrationFingerprint
+
+@Patch(description = "Filter what will be shown in the timeline.", requiresIntegrations = true)
+object TimelineFilterPatch : BytecodePatch(
+    setOf(TimelineConstructorFingerprint, TimelineFilterIntegrationFingerprint, PostsResponseConstructorFingerprint)
+) {
+    /**
+     * Add a filter to hide the given timeline object type.
+     * The list of all Timeline object types is found in the TimelineObjectType class,
+     * where they are mapped from their api name (returned by tumblr via the HTTP API) to the enum value name.
+     *
+     * @param typename The enum name of the timeline object type to hide.
+     */
+    @Suppress("KDocUnresolvedReference")
+    internal lateinit var addObjectTypeFilter: (typename: String) -> Unit private set
+
+    override fun execute(context: BytecodeContext) {
+
+        TimelineFilterIntegrationFingerprint.result?.let { integration ->
+            val startIndex = integration.scanResult.patternScanResult!!.startIndex
```

Comment:
Maybe give this a better variable name. Something that describes what the current index in the code is.